File: tests/fixtures/csharp/type_inference/control_flow/control_flow.cs
using System;
using System.Collections.Generic;

public class ControlFlowDemo
{
    public static string HandleIsCheck(object obj)
    {
        if (obj is string s)
        {
            return s.ToUpper();
        }
        else if (obj is int n)
        {
            return $"number: {n}";
        }
        return "unknown";
    }

    public static string HandleTryCatch()
    {
        try
        {
            int.Parse("not_a_number");
            return "parsed";
        }
        catch (FormatException e)
        {
            return $"error: {e.Message}";
        }
        catch (Exception e)
        {
            return $"general error: {e.Message}";
        }
    }

    public static string HandleTryCatchWhen()
    {
        try
        {
            throw new ArgumentException("test");
        }
        catch (ArgumentException ex) when (ex.Message.Contains("test"))
        {
            return $"filtered: {ex.Message}";
        }
    }

    public static string HandleTryFinally()
    {
        string result = "default";
        try
        {
            result = "modified";
        }
        finally
        {
            result = result + " (finally)";
        }
        return result;
    }

    public static string HandleMultipleCatch()
    {
        try
        {
            throw new InvalidOperationException("test");
        }
        catch (InvalidOperationException e)
        {
            return $"invalid op: {e.Message}";
        }
        catch (ArgumentException e)
        {
            return $"arg: {e.Message}";
        }
        catch (Exception e)
        {
            return $"general: {e.Message}";
        }
    }

    public static void Main()
    {
        Console.WriteLine(HandleIsCheck("hello"));
        Console.WriteLine(HandleIsCheck(42));
        Console.WriteLine(HandleTryCatch());
        Console.WriteLine(HandleTryCatchWhen());
        Console.WriteLine(HandleTryFinally());
        Console.WriteLine(HandleMultipleCatch());
    }
}

File: tests/fixtures/go/type_inference/control_flow/control_flow.go
package main

import (
	"errors"
	"fmt"
	"strconv"
)

func handleError(err error) string {
	if err != nil {
		return err.Error()
	}
	return "no error"
}

func divide(a, b int) (int, error) {
	if b == 0 {
		return 0, errors.New("division by zero")
	}
	return a / b, nil
}

func parseAndProcess(s string) (int, error) {
	n, err := strconv.Atoi(s)
	if err != nil {
		return 0, fmt.Errorf("parse error: %w", err)
	}
	if n < 0 {
		return 0, errors.New("negative number")
	}
	return n * 2, nil
}

func processTwoErrors(a, b string) (int, error) {
	n1, err := strconv.Atoi(a)
	if err != nil {
		return 0, err
	}
	n2, err := strconv.Atoi(b)
	if err != nil {
		return 0, err
	}
	return n1 + n2, nil
}

func main() {
	err := errors.New("test error")
	fmt.Println(handleError(err))
	fmt.Println(handleError(nil))

	result, err := divide(10, 3)
	if err != nil {
		fmt.Println("error:", err)
	} else {
		fmt.Println("result:", result)
	}

	parsed, err := parseAndProcess("42")
	if err != nil {
		fmt.Println("error:", err)
	} else {
		fmt.Println("parsed:", parsed)
	}
}
